.github/workflows/dotnet3.1Build.yml

```diff
@@ -0,0 +1,38 @@
+name: CodingGuidelines Build
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  schedule:
+    - cron: 0 6 * * 1
+env:
+  version: 0.1.9
+jobs:
+  build:
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: microsoft/setup-msbuild@v1.0.2
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 3.1.x
+    - name: Use latest NuGet
+      uses: nuget/setup-nuget@v1
+    - name: Get Directory
+      run: pwd
+    - name: Dotnet Restore dependencies
+      run: dotnet restore
+    - name: Build
+      run: dotnet build --no-restore
+    - name: Test
+      run: dotnet test --no-build --verbosity normal
+    - name: dotnet pack
+      run: dotnet pack -p:Version="${{ env.version }}-ci-${{ github.run_id }}" -o ${{ github.workspace }}
```

Comment:
I believe we want to use run_number rather than run_id since run_id will be random and versioning ordering will not be consistent. Alternatively date based would work as well.